src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -356,17 +368,24 @@ fn partition(
 }
 
 fn healthcheck(config: CloudwatchLogsSinkConfig) -> Result<super::Healthcheck, String> {
+    if config.group_name.is_dynamic() {
+        info!("cloudwatch group_name is dynamic; skipping healthcheck.");
+        return Ok(Box::new(future::ok(())));
+    }
+
+    let group_name = String::from_utf8_lossy(&config.group_name.get_ref()[..]).into_owned();
```

Comment:
It seems like this is the only place we use `get_ref`, which makes me think we should replace it with `as_str`.

Reply:
Yeah, I'm not convinced str is any better, I kinda viewed get_ref as a get a reference to whatever this thing is holding kinda. I could also add a `as_str`.

Another argument for as_str is that we only have From<String> really so that would make sense. Maybe `AsRef<str>` works better?

Reply:
My point was just that we already have a string representation available and could lose the `from_utf8_lossy().into_owned()`.

My worry with `AsRef<str>` would be making it easy to accidentally use as a string without rendering, but I guess that's also valid for `as_str`. Maybe the name should make it clear you're getting the unrendered inner value? Like `as_raw_str` or `as_inner` or something like that? Not a huge deal either way.

Reply:
This is actually why I like `get_ref` since it indicates that you are getting the ref to the inner value. Inspiration for this name actually comes from the futures library and other rust code. https://docs.rs/futures/0.1.28/futures/stream/struct.Filter.html#method.get_ref They like to use get_ref, get_mut, and into_inner.